BD-1494: Canvas entry properties and event properties (#3912)
braze-inc/braze-docs: _docs/_user_guide/engagement_tools/canvas/create_a_canvas/canvas_entry_properties_event_properties.md

---
nav_title: Canvas Entry Properties and Event Properties
article_title: Canvas Entry Properties and Event Properties
page_type: reference
description: "This reference article describes the differences between Canvas entry properties and event properties, and when to use each property."
tool: Canvas
page_order: 8
---

# Canvas entry properties and event properties

> This reference article covers information about `canvas_entry_properties` and `event_properties`, including when to use each property and the differences in behavior.

Though similar in name, Canvas entry properties and event properties function differently within your Canvas workflows. Properties of events or API calls that trigger a user's entry into a Canvas are known as `canvas_entry_properties`. Properties of events that occur as a user moves through a Canvas journey are known as `event_properties`. The key difference here is `canvas_entry_properties` focuses on more than just events by also accessing the properties of entry payloads in API-triggered Canvases.

> Note that behavior can vary across workflows built with the original Canvas editor or with Canvas Flow. For example, in the original Canvas editor, you can use `event_properties` in the first Full Step if it's an action-based step. In Canvas Flow, Full Steps aren't supported, so this does not apply. 

Reviewer: ```suggestion
Behavior can vary between workflows built with the original Canvas editor versus Canvas Flow. For example, in the original Canvas editor, you can use `event_properties` in the first Full Step if it's an action-based step. In Canvas Flow, Full Steps aren't supported, so this does not apply. 
```

| Persistence | Can be referenced by all [Message][1] steps for the duration of a Canvas for Canvas Flow only. | - Can only be referenced once. <br> - Cannot be referenced by any subsequent Message steps. |
| Original Canvas behavior | Can only reference `canvas_entry_properties` in the first Full Step of a Canvas. The Canvas must be action-based or API triggered. | You can reference `event_properties` in any Full Step that uses action-based delivery in a Canvas. However, if a user is using a [Canvas component][2] in the original Canvas editor, the behavior follows the Canvas Flow rules for `event_properties`. |
| Canvas Flow behavior | Can reference `canvas_entry_properties` in any step of a Canvas. | - Can reference `event_properties` in the first Message step **after** an [Action Paths][3] step where the action taken is a custom event or purchase event. <br> - Cannot be after the Everyone Else path of the Action Paths step. <br> - Can have other non-Message Canvas components in between the Action Path and Message Step. If one of these non-Message Canvas components is an Action Paths step, the user can go through that Action Paths' Everyone Else path. | 

> {: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

Reviewer: Minor style thing but you could make this table have both a header row and column by referencing this code for the [SMS short and long codes](https://github.com/braze-inc/braze-docs/blob/develop/_includes/archive/short_and_long_codes.md#:~:text=%3Cstyle%20type%3D%22text,vertical%2Dalign%3Atop%7D%20%3C/style%3E)

The `canvas_entry_properties` are configured in the [Entry Schedule][4] step of creating a Canvas and will correspond to when a user enters a Canvas. These `canvas_entry_properties` can also be referenced in any Message Step in Canvas Flow since Canvas Flow supports [persistent entry properties][5]. 

In this Canvas, we have a user journey that begins at an Action Paths step to determine if a user has added an item to their wishlist. From here, if the user has added an item, then they will experience a delay before receiving a message "New item in your wishlist!" from the Message step. The first Message step in a user journey will have access to the custom `event_properties` from your Action Paths step, so in this case, we're able to include `` {% raw %} {{event_properties.${property_name}}} {% endraw %}`` here in this Message step as part of our message content. 

If a user doesn't add an item to their wishlist, they go through the Everyone Else path, so the `event_properties` can't be referenced and will reflect an invalid settings error.

> Note that you’ll only have access to `event_properties` if your Message Step can be traced back to a non-Everyone Else path in an Action Path step. If the Message Step is connected to an Everyone Else path but can be traced back to an Action Paths step, then you will also still have access to `event_properties`. For more information on these behaviors, check out [Message][8].

Reviewer: > If the Message Step is connected to an Everyone Else path but can be traced back to an Action Paths step, then you will also still have access to `event_properties`.

I don't understand that bit 🤔
